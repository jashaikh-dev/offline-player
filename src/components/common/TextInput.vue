
<template>
    <div class="text-input-container">
        <input 
            class="text-input" 
            :type="type" 
            :placeholder="placeholder"
            :name="name"
            :id="id"
            @input="onInput"
        />
    </div>
</template>

<script>
    export default {
        name: 'TextInput',
        props: ["type", "placeholder", "name", "id", "onTextChange"],
        methods: {
            onInput(event){
                this.$emit('onTextChange', event)
            }
        }
    }
</script>

<style lang="less" scoped>
.text-input-container {
    height: 50px;
    padding: 2px;
    width: 80%;
    & > .text-input {
        width: 80%;
        height: 40px;
        border: none;
        border-radius: 2px;
        padding: 2px;
        padding-left: 5px;

    }
}
</style>