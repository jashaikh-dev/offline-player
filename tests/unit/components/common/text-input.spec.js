import { shallowMount } from '@vue/test-utils'
import TextInput from '@/components/common/TextInput'

describe('TextInput.spec.js', () => {

    let wrapper
    beforeEach(() => {
        wrapper = shallowMount(TextInput)
    })

    it('should match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})