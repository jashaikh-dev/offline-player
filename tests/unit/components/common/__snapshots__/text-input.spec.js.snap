// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput.spec.js should match snapshot 1`] = `<div class="text-input-container"><input class="text-input"></div>`;
